jupyterhub/app.py

```diff
@@ -1258,6 +1258,21 @@ def add_url_prefix(prefix, handlers):
         """
     ).tag(config=True)
 
+    user_redirect_hook = Callable(
+        None,
+        allow_none=True,
+        help="""
+        Callable to affect behavior of /user-redirect/
+
+        A callable that receives two parameters - a handler and the path passed to
+        `/user-redirect/<path>`, and returns a URL to redirect the user to. This
+        can be used to customize how the `user-redirect` URL works.
+
+        To get the default behavior of /user-redirect/ leave this property unset
+        or return None from your callable.
+        """
+    ).tag(config=True)
```

Comment:
Will `path` include the query parameters as well? I become uncertain, because of the terminology when parsing an url gives you `.path` and `.query` for example.

```
    path = urlparse(r.url).path
    assert path == ujoin(app.base_url, '/hub/login')
    query = urlparse(r.url).query
```

Reply:
Yep, Path didn't contain query parameters. I'm just passing in the request object directly now, although I might have to pass in the handler too.